app/scripts/controllers/permissions/specifications.js

```diff
@@ -181,25 +189,30 @@ export const getPermissionSpecifications = ({
  * corresponds to a PreferencesController identity.
  *
  * @param {string[]} accounts - The accounts associated with the caveat.
- * @param {() => Record<string, Identity>} getIdentities - Gets all
- * PreferencesController identities.
+ * @param {() => Record<string, import('@metamask/keyring-api').InternalAccount>} getInternalAccounts -
+ * Gets all AccountsController InternalAccounts.
  */
-function validateCaveatAccounts(accounts, getIdentities) {
+function validateCaveatAccounts(accounts, getInternalAccounts) {
   if (!Array.isArray(accounts) || accounts.length === 0) {
     throw new Error(
       `${PermissionNames.eth_accounts} error: Expected non-empty array of Ethereum addresses.`,
     );
   }
 
-  const identities = getIdentities();
+  const internalAccounts = getInternalAccounts();
   accounts.forEach((address) => {
     if (!address || typeof address !== 'string') {
       throw new Error(
         `${PermissionNames.eth_accounts} error: Expected an array of Ethereum addresses. Received: "${address}".`,
       );
     }
 
-    if (!identities[address]) {
+    if (
+      !internalAccounts.find(
```

Comment:
Nit: `some` can be used if you don't care about the result:

``` suggestion
      !internalAccounts.some(

Reply:
Solved in [b5e5235](https://github.com/MetaMask/metamask-extension/pull/21437/commits/b5e52351760fc47bbf5e66915cb819b356267c71)